pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -320,12 +315,22 @@ func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user st
 		// Zero flags argument means that the authorized_keys file is overwritten with the authorizedKeys
 		return domain.AuthorizedSSHKeysSet(user, authorizedKeys, 0)
 	}()
-	if errors.Is(err, libvirt.ERR_NO_SUPPORT) {
-		// If AuthorizedSSHKeysSet method is not supported, use the old method
-		desiredAuthorizedKeys := strings.Join(authorizedKeys, "\n")
-		return l.agentWriteAuthorizedKeysFile(domName, user, desiredAuthorizedKeys)
+	if firstErr == nil {
```

Comment:
Is there a way to assert on certain errors?

Reply:
Which errors could be expected?

Reply:
The `libvirt.ERR_INTERNAL_ERROR` is returned if guest agent does not support the command. But I in my opinion it should be safe to not assert an error.
Do you think it is better to check only for certain errors?

Reply:
If there's more to distinguish than just an internal error we should check it IMO.

Reply:
Ok, and what should we check? Do you mean that on some errors we don't try the old way of setting ssh keys?

Reply:
Which errors can be returned? Only `libvirt.ERR_INTERNAL_ERROR` or also others? Depending on that I would decide if it's an error we can recover from (and continue with the old way) or if we should abort immediately.

Reply:
Ok, I will look into it. These errors are directly propagated from libvirt. I will read it's source in more detail.

Reply:
I think it is too tedious to go through all code paths to see what errors can be returned.
I would rather keep this as is.

Reply:
OK

Reply:
Thinking about this. Can you add a log here so at least it becomes obvious what happens from looking at the log files?

Reply:
done